addons/mail/static/src/core/common/composer.xml

```diff
@@ -32,14 +32,17 @@
             <div t-if="showComposerAvatar" class="o-mail-Composer-sidebarMain flex-shrink-0 d-flex">
                 <img class="o-mail-Composer-avatar mx-auto o_avatar rounded-3" t-att-src="(thread?.effectiveSelf or message.effectiveSelf).avatarUrl" alt="Avatar of user"/>
             </div>
-            <div class="o-mail-Composer-coreHeader text-truncate small p-2" t-if="props.composer.replyToMessage">
-                <span class="cursor-pointer" t-on-click="() => env.messageHighlight?.highlightMessage(props.composer.replyToMessage, props.composer.thread)">
-                    Replying to <b t-esc="props.composer.replyToMessage.authorName"/>
-                </span>
-                <span t-if="props.composer.replyToMessage.thread?.notEq(props.composer.thread)">
-                    on: <b><t t-esc="props.composer.replyToMessage.thread.displayName"/></b>
-                </span>
-                <i class="fa fa-lg fa-times-circle rounded-circle p-0 ms-1 cursor-pointer" title="Stop replying" t-on-click="() => (props.composer.replyToMessage = undefined)"/>
+            <div class="o-mail-Composer-coreHeader">
+                <div t-if="props.composer.replyToMessage" class="text-truncate small p-2">
+                    <span class="cursor-pointer" t-on-click="() => env.messageHighlight?.highlightMessage(props.composer.replyToMessage, props.composer.thread)">
+                        Replying to <b t-esc="props.composer.replyToMessage.authorName ?? props.composer.replyToMessage.email_from"/>
```

Comment:
`authorName` has already `email_from` fallback under the hood, no need for this.